intellij/src/main/java/toolWindow/MyToolWindow.java

```diff
@@ -77,18 +90,76 @@ public JComponent getInput() {
         JButton generateButton = new JButton("Generate");
-        generateButton.addActionListener(new ActionListener() {
+        JProgressBar progressBar = new JProgressBar();
+        progressBar.setIndeterminate(true);
+        JPanel generateButtonPanel = new JPanel(new FlowLayout(FlowLayout.CENTER));
+        generateButtonPanel.add(generateButton); // Add the generate button to the panel
+
+
+        progressBarPanel.setLayout(new FlowLayout(FlowLayout.CENTER)); // changed this
+        progressBar.setName("generate");
+        progressBarPanel.add(progressBar);
+        generateButtonPanel.add(progressBarPanel);
+        progressBar.setVisible(false);
+        generateButton.addActionListener(new ActionListener() {  // generatebutton action listener
             @Override
             public void actionPerformed(ActionEvent e) {
-                String descriptionPath = yamlFilePathField.getText();
-                String output = outputpath.getText();
-                String clientClass = clientClassField.getText();
-                String clientClassNamespace = postClientNameField.getText();
-                KiotaGenerationLanguage selectedLanguage = (KiotaGenerationLanguage) languageComboBox.getSelectedItem();
-                generateClientHandler.generateclient(descriptionPath, output, selectedLanguage, include, exclude, clientClass, clientClassNamespace);
+                generateButton.setVisible(false);
+                progressBar.setVisible(true);
+                progressBarPanel.removeAll(); // Clear any previous components
+                progressBarPanel.add(progressBar); // Add the progress bar
+
+                // perform the generation in a separate thread
+                SwingWorker<Void, Void> worker = new SwingWorker<Void, Void>() {
+                    @Override
+                    protected Void doInBackground() throws Exception {
+                        String descriptionPath = yamlFilePathField.getText();
+                        String output = outputpath.getText();
+                        String clientClass = clientClassField.getText();
+                        String clientClassNamespace = postClientNameField.getText();
+                        KiotaGenerationLanguage selectedLanguage = (KiotaGenerationLanguage) languageComboBox.getSelectedItem();
+
+                        generateClientHandler.generateclient(descriptionPath, output, selectedLanguage, include, exclude, clientClass, clientClassNamespace);
+
+                        return null;
+                    }
+                    @Override
+                    protected void done() {
+                        // Generation is complete, re-enable the button
+                        progressBarPanel.setVisible(true);
+                        Object response = generateClientHandler.getResponse();
+                        String responseString = response.toString();
```

Comment:
What is the structure of the response string? If it has structure, we may want to create a model so that we can use a strongly-typed object for parsing the results. In this case, since it is a response for the `generate` command, we could consider a name like `KiotaGenerateResponse`

We should take a look at the structure of the response.

Reply:
tracking #3168